File: examples/kotlin/UserService.kt
import java.security.MessageDigest
import java.sql.DriverManager
import java.net.URL
import java.io.Serializable

class UserService {
    
    fun getUserProfile(userId: String, currentUser: String) {
        // No authorization check
        fetchUserData(userId)
    }
    
    fun hashPassword(password: String): String {
        // Weak hashing algorithm
        val md = MessageDigest.getInstance("MD5")
        val digest = md.digest(password.toByteArray())
        return digest.joinToString("") { "%02x".format(it) }
    }
    
    fun findUser(username: String) {
        val connection = DriverManager.getConnection("jdbc:mysql://localhost/db", "user", "pass")
        val statement = connection.createStatement()
        // SQL injection vulnerability
        val query = "SELECT * FROM users WHERE username = '$username'"
        statement.executeQuery(query)
    }
    
    fun processOrder(orderData: String) {
        // No input validation
        executeOrder(orderData)
    }
    
    fun initializeApp() {
        // Debug mode enabled
        System.setProperty("debug", "true")
        val apiKey = "sk-1234567890abcdef"
    }
    
    fun authenticateUser(username: String, password: String): Boolean {
        // Weak authentication
        return username == "admin" && password.length > 6
    }
    
    fun loadUserData(serializedData: String): Any {
        // Unsafe operations
        return serializedData
    }
    
    fun logUserActivity(username: String, sessionToken: String) {
        // Logging sensitive information
        println("User $username session: $sessionToken")
    }
    
    fun fetchExternalResource(url: String) {
        // SSRF vulnerability
        val connection = URL(url).openConnection()
        connection.getInputStream()
    }
    
    fun executeCommand(command: String) {
        // Command injection
        val process = ProcessBuilder("sh", "-c", command).start()
        process.waitFor()
    }
    
    private fun fetchUserData(userId: String) {}
    private fun executeOrder(data: String) {}
}

File: examples/ruby/user_service.rb
require 'digest'
require 'mysql2'
require 'net/http'
require 'json'

class UserService
  
  def get_user_data(user_id, current_user)
    # No authorization check
    fetch_user(user_id)
  end
  
  def hash_password(password)
    # Weak hashing
    Digest::MD5.hexdigest(password)
  end
  
  def find_user(username)
    client = Mysql2::Client.new(host: "localhost", username: "root", password: "password")
    # SQL injection
    query = "SELECT * FROM users WHERE username = '#{username}'"
    client.query(query)
  end
  
  def process_order(order_data)
    # No input validation
    execute_order(order_data)
  end
  
  def setup_environment
    # Debug mode enabled
    ENV['DEBUG'] = 'true'
    @api_key = 'sk-1234567890abcdef'
  end
  
  def authenticate(username, password)
    # Weak authentication
    username == 'admin' && password.length > 6
  end
  
  def load_user_data(serialized_data)
    # Unsafe deserialization
    eval(serialized_data)
  end
  
  def log_user_activity(username, password)
    # Logging sensitive data
    puts "Login: #{username} with password: #{password}"
  end
  
  def fetch_external_data(url)
    # SSRF vulnerability
    uri = URI(url)
    Net::HTTP.get(uri)
  end
  
  def execute_command(command)
    # Command injection
    system(command)
  end
  
  private
  
  def fetch_user(user_id)
    # Implementation
  end
  
  def execute_order(data)
    # Implementation
  end
end